massa-ledger-worker/src/ledger_db.rs

```diff
@@ -266,38 +275,42 @@ impl LedgerDB {
     /// * The last taken key (this is an optimization to easily keep a reference to the last key)
     pub fn get_ledger_part(
         &self,
-        cursor: StreamingStep<Vec<u8>>,
-    ) -> Result<(Vec<u8>, StreamingStep<Vec<u8>>), ModelsError> {
+        cursor: StreamingStep<Key>,
+    ) -> Result<(Vec<u8>, StreamingStep<Key>), ModelsError> {
         let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
         let opt = ReadOptions::default();
         let ser = VecU8Serializer::new();
-        let key_serializer = KeySerializer::new();
+        let key_deserializer = KeyDeserializer::new(self.max_datastore_key_length);
```

Comment:
The deserializer could be in self to avoid multiple creations each time we call this.